app/javascript/components/upload/UploadWizard.js

```diff
@@ -171,10 +178,24 @@ export function RawUploadWizard({ studyAccession, name }) {
     updateFile(fileId, { saveProgress: overallCompletePercent })
   }
 
+  /** cancels the pending upload and deletes the file */
+  async function cancelUpload(requestCanceller) {
+    requestCanceller.cancel()
+    const fileId = requestCanceller.fileId
+
+    deleteFileFromForm(fileId)
+    // wait a brief amount of time to minimize the timing edge case where the user hits 'cancel' while the request
+    // completed.  In that case the file will have been saved, but we won't have the new id yet.  We need to wait until
+    // the id comes back so we can address the delete request properly
+    // We also don't await the delete since if it errors, it is likely because the file never got created or
+    // got deleted for other reasons (e.g. invalid form data) in the meantime
+    setTimeout(() => deleteFileFromServer(requestCanceller.fileId), 500)
```

Comment:
I think this is causing the behavior I'm seeing:

1. Click `Save & Upload` on a large file
2. Wait a few seconds
3. Click `Cancel`
4. Dialog shows briefly, then automatically closes itself
5. Click `Cancel` again, and dialog stays open

I get that we have an edge case here regarding whether or not the file persisted, but the timeout is resulting in a weird UX.

Reply:
yeah, I wonder if the dialog close happens if you happen to have the dialog open during the exact moment when an individual chunk completes.  I'll take a look

Reply:
yup, confirmed that that's the problem -- if you have the cancel dialog open during the time when the first chunk resolves, it clears itself.  This is because the file gets updated with the new id from the server, and we use the id as the `key` property for the file list components.  So when the key changes, it gets treated as a new component, and the cancel modal open/close state gets reset.

I'm going to open a separate ticket for this, because it's a non-trivial fix to change how we're id'ing all the components.  But it's an important change since this id change issue will likely come up in other areas too.